security/advancedtls/advancedtls_test.go

```diff
@@ -47,6 +45,94 @@ func Test(t *testing.T) {
 	grpctest.RunSubTests(t, s{})
 }
 
+// certStore contains all the certificates used in the integration tests.
```

Comment:
Making them to a separate package would require us to make all the structs and loading functions exportable, and hence they will become part of the public API. These functions are just helpers for tests, so I'd like to avoid exporting them. Can we keep them in the same package?
The discussion raised my another question as well: should the tests be in a separate package, or in the same package as the main code? Currently I could see both ways exist across the repo. Are there any rationale behind when choosing the two?

Reply:
You can make them a separate package under `security/advancedtls/internal/testutils`. Packages under `internal` are not part of the public API.

Regarding whether tests need to be in a separate package or the same package, I think it depends. Most of the time, unit tests will be in the same package. Under some conditions though, one might want to write a test which does not have access to non-exported symbols in the package (some form of an e2e test). These tests live in a separate package. And sometimes we put tests in a separate package to avoid some kind of circular dependency.

Reply:
Thanks for the explanation! That makes sense. I will move the loading function to an internal package.